conans/test/unittests/model/test_conan_file.py

```diff
@@ -0,0 +1,49 @@
+# coding=utf-8
+
+import os
+import shutil
+import tempfile
+import textwrap
+import unittest
+import uuid
+
+from conans.client.graph.python_requires import ConanPythonRequire
+from conans.client.loader import ConanFileLoader, ProcessedProfile
+from conans.client.tools import save
+from conans.model.conan_file import is_alias_conanfile
+
+
+class IsAliasConanfileTest(unittest.TestCase):
+    conanfile_base = textwrap.dedent("""\
+        from conans import ConanFile
+        
+        class A(ConanFile):
+            {placeholder}
+        """)
+    conanfile_alias = conanfile_base.format(placeholder='alias = "whatever"')
+    conanfile_not_alias = conanfile_base.format(placeholder='pass')
+
+    def run(self, *args, **kwargs):
+        self.tmp_folder = tempfile.mkdtemp()
+        try:
+            super(IsAliasConanfileTest, self).run(*args, **kwargs)
+        finally:
+            shutil.rmtree(self.tmp_folder)
+
+    def _load_conanfile(self, content):
```

Comment:
Maybe use the `MockedConanFile` here or do you want to really test the load part?

Reply:
`MockConanfile` is not an instance of `ConanFile`... although there is a `ConanFileMock` that it is (although it is not calling the `super().__init__` (feel suspicious about it).